### PR TITLE
Feature/issue 65

### PR DESCRIPTION
**Pull request by edouardmichelin**

### CODE REVIEW

**Comment by edouardmichelin** [comment]
Cannot do more than 79% coverage for now

File: app/src/androidTest/java/com/studhub/app/NavigationTest.kt [hunk 1]
```diff
@@ -81,6 +81,31 @@ class NavigationTest {
...
+
+        // click the save button
+        composeTestRule
+            .onNodeWithText(str(R.string.profile_edit_form_btn_save))
+            .performScrollTo()
+            .performClick()
+
+        // assert we are back on the profile page
+        composeTestRule.onNodeWithText(str(R.string.profile_title))
+    }
```
Comment (NervzZ) [inline app/src/androidTest/java/com/studhub/app/NavigationTest.kt:107]: You forgot to test here

File: app/src/main/java/com/studhub/app/domain/usecase/user/UpdateCurrentUserInfo.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import com.studhub.app.domain.repository.UserRepository
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+/**
+ * Use case for updating the current user with given profile information in [userRepository]
+ *
+ * @param [userRepository] the repository which the use case will act on
+ * @param [authRepository] the repository which the use case will retrieve the logged in user from
+ * @
```
Comment (NervzZ) [inline app/src/main/java/com/studhub/app/domain/usecase/user/UpdateCurrentUserInfo.kt:15]: Delete this

**Review by NervzZ — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                            )
+                        )
+
+                        navigateToProfile()
+                    }
+                )
+            }
+        }
+    }
+}
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt:70]: Small preview function?

File: app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import com.studhub.app.presentation.ui.common.text.BigLabel
+
+@Composable
+fun EditProfileScreen(
+    viewModel: ProfileViewModel = hiltViewModel(),
+    navigateToProfile: () -> Unit
+) {
+    when (val currentUser = viewModel.currentUser) {
+        is ApiResponse.Loading -> LoadingCircle()
+        is ApiResponse.Failure -> {}
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt:28]: handle failure or not?

File: app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                    .horizontalScroll(scrollState),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Spacer(modifier = Modifier.height(48.dp))
+
+                BigLabel(stringResource(R.string.profile_edit_title))
+
+                Spacer(modifier = Modifier.height(48.dp))
+
+                EditProfileForm(
```
Comment (raynasr) [inline app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt:49]: EditProfileForm(
    // ...
    onSubmit = {
        viewModel.updateUserInfo(
            User(
                // ...
            )
        ).onEach { result ->
            when (result) {
                is ApiResponse.Loading -> 
                    CircularProgressIndicator()
                }
                is ApiResponse.Failure -> {
                    // handle or not?
                }
                is ApiResponse.Success -> {
                    navigateToProfile()
                }
            }
        }.launchIn(viewModelScope)
    }
)

The point is to display a loading indicator while the user info is being updated (it replaces the form).
.onEach returns a flow that we handle with success, failure and loading

**Review by raynasr — COMMENTED** [review]
Looking forward to hearing your thoughts on my comments before proceeding.

File: app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt (hunk 4 above)
Comment (edouardmichelin) [reply app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt:28]: We haven't decided on a way to handle errors yet so for now this is not handled but can easily be in the future

File: app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                    firstName = firstName,
+                    lastName = lastName,
+                    userName = userName,
+                    phoneNumber = phoneNumber,
```
[the remark this answers is not in the document]
Comment (edouardmichelin) [reply app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt:53]: For the moment there is no validation whatsoever (on this form or any other).

File: app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt (hunk 5 above)
Comment (edouardmichelin) [reply app/src/main/java/com/studhub/app/presentation/profile/EditProfileScreen.kt:49]: For now these views are more placeholders than anything else.
I think waiting for completion before navigating back to the profile page is a good idea though.

**Review by raynasr — APPROVED** [review]
Discussed changes, approved.

**Review by NervzZ — APPROVED** [review]
Looks good now after requested changes. Nice to keep up with the documentation and test coverage threshold

**Pull request merged**